com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -22,29 +22,16 @@ public sealed class NetworkObject : MonoBehaviour
     {
         [HideInInspector]
         [SerializeField]
-        internal ulong GlobalObjectIdHash64;
+        internal ulong GlobalObjectIdHash;
 
 #if UNITY_EDITOR
         private void OnValidate()
         {
             var globalObjectIdString = UnityEditor.GlobalObjectId.GetGlobalObjectIdSlow(this).ToString();
-            GlobalObjectIdHash64 = XXHash.Hash64(globalObjectIdString);
-
-            // Set this so the hash can be serialized on Scene objects. For prefabs, they are generated at runtime.
-            ValidateHash();
+            GlobalObjectIdHash = XXHash.Hash64(globalObjectIdString);
         }
 #endif
 
-        internal void ValidateHash()
-        {
-            if (string.IsNullOrEmpty(PrefabHashGenerator))
```

Comment:
Per my comment, assuming there is in fact a reason a user might want to force-choose a hash, we could change the default to be null, but still allow them to "expert mode" override it

Reply:
I have 2 questions:

1. what is the user-need?
2. should we serve that need with `PrefabHashGenerator` string?

to me, even if the user-need is justified and we think it's important enough and valid to keep around, I wouldn't serve/satisfy that user-need with `PrefabHashGenerator` string implementation — it'd be an implementation on its own, relying on top of this infrastructure.

Reply:
I too would be curious to know what the use case for having two different prefabs that translate to the same hash but could be different assets?  _(the last part I am assuming would be the only scenario to consider for this condition to be useful)_ 

The only scenario where this could be useful (IMO) would be to have different assets per platform (i.e. mobile vs console typically have vast deltas between assets).  Under this scenario, I could see where being able to directly assign the Prefab hash could be useful if you had no other "good options" that were minimal effort for the desired result.

If this is the only reason why having two different prefabs with the same prefab hash, then this might be another "pro [Addressables](https://unity.com/how-to/simplify-your-content-management-addressables)" scenario that could solve for this type of situation.

Otherwise, for a near future solution I am sure we could figure out some pattern to handle having a way to "link GlobalObjectIdHash" values together.  Maybe something like a new MLAPI aware component (i.e. "NetworkPrefabLink") that could be applied to a GameObject inside of a "master prefab" asset.  You could then add as many prefabs as you wanted as children under the root GameObject containing the NetworkPrefabLink component.  From there, linking the prefabs and the logic to determine which prefab to use could be completely customizable by the user.  The GlobalObjectIdHash of the "master prefab" would then become the GlobalObjectIdHash for any of the prefabs defined within it.

That is one way we could approach any potential breaking features/intended functionality while also providing a more expansive/modular pattern to the users at the same time.   :)

Reply:
I'm seeing 3 completely separate features here:

1. Network Object
2. Network Prefab
3. Network Addressable

we're going to have the first 2 and leave the 3rd for later.